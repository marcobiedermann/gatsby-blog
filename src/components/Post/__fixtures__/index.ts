import { PostProps } from '..';

const post: PostProps = {
  date: '2000-01-01',
  html: '<p>Lorem Ipsum</p>',
  tags: ['tag-1'],
  title: 'First Post',
  timeToRead: 1,
};

export default post;
