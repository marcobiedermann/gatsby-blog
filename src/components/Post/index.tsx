/* eslint-disable react/no-danger */

import dayjs from 'dayjs';
import React from 'react';
import { DATE_FORMAT } from '../../constants/date';
import Tags from '../Tags';

export interface PostProps {
  date: string;
  html: string;
  tags: string[];
  timeToRead: number;
  title: string;
}

function Post(props: PostProps) {
  const { date, html, tags, timeToRead, title } = props;

  return (
    <article>
      <header>
        <h1>{title}</h1>
        <div>
          <time dateTime={date}>{dayjs(date).format(DATE_FORMAT)}</time> â€¢ {timeToRead} mins.
        </div>
      </header>
      <main>
        <div dangerouslySetInnerHTML={{ __html: html }} />
      </main>
      <footer>
        <Tags tags={tags} />
      </footer>
    </article>
  );
}

export default Post;
