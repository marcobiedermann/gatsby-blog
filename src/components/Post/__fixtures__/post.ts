import { PostProps } from '../Post';
import tags from '../../Tags/__fixtures__/tags';

const post: PostProps = {
  date: '2000-01-01',
  html: '<p>Lorem Ipsum</p>',
  tags: tags.tags,
  title: 'First Post',
  timeToRead: 1,
};

export default post;
