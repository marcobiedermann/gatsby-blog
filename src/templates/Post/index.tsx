import { graphql, PageProps } from 'gatsby';
import React, { FC } from 'react';
import Button from '../../components/Button';
import Layout from '../../components/Layout';
import Pagination from '../../components/Pagination';
import Post from '../../components/Post';
import Section from '../../components/Section';

export interface DataType {
  markdownRemark: {
    frontmatter: {
      date: string;
      tags: string[];
      title: string;
    };
    html: string;
    timeToRead: number;
  };
}

interface PageContextType {
  id: string;
  next: {
    fields: {
      slug: string;
    };
    id: string;
  } | null;
  previous: {
    fields: {
      slug: string;
    };
    id: string;
  } | null;
}

const PostTemplate: FC<PageProps<DataType, PageContextType>> = (props) => {
  const {
    data: {
      markdownRemark: {
        frontmatter: { date, tags, title },
        html,
        timeToRead,
      },
    },
    pageContext: { next, previous },
  } = props;

  return (
    <Layout>
      <Section>
        <Post date={date} html={html} tags={tags} timeToRead={timeToRead} title={title} />
      </Section>
      <Section>
        <Pagination next={next} previous={previous} />
        <p>
          <Button to="/">back</Button>
        </p>
      </Section>
    </Layout>
  );
};

export const query = graphql`
  query($id: String!) {
    markdownRemark(id: { eq: $id }) {
      frontmatter {
        date
        tags
        title
      }
      html
      timeToRead
    }
  }
`;

export default PostTemplate;
