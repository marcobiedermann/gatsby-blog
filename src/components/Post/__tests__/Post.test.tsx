import { render } from '@testing-library/react';
import React from 'react';
import Post from '..';
import post from '../__fixtures__/post';

describe('post component', () => {
  it('should render correctly', () => {
    const { container } = render(<Post {...post} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
