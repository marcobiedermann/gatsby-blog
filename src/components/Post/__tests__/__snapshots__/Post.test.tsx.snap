// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post component should render correctly 1`] = `
<article>
  <header>
    <h1>
      First Post
    </h1>
    <div>
      <time
        datetime="2000-01-01"
      >
        01 Jan. 2000
      </time>
       â€¢ 
      1
       mins.
    </div>
  </header>
  <main>
    <div>
      <p>
        Lorem Ipsum
      </p>
    </div>
  </main>
  <footer>
    <ul
      class="tags"
    >
      <li>
        <a
          class="tag"
          href="/tags/tag-1"
        >
          tag-1
        </a>
      </li>
    </ul>
  </footer>
</article>
`;
